.github/workflows/test-gaia-v14-stateful-upgrade-steps.yml

```
@@ -0,0 +1,359 @@
+          sudo apt dist-upgrade -y
+      - name: Install openssh-server
+        run: |
+          sudo apt install -y openssh-server
+          sudo mkdir /run/sshd
+          sudo /usr/sbin/sshd
+      - name: Setup SSH auth
+        run: |
+          mkdir -m 700 ~/.ssh
+          echo "$SSH_KEYS" > ~/.ssh/authorized_keys
+      - name: start-version
+        run: echo ${{inputs.gaia_version}}
+      - name: target-version
+        run: echo ${{inputs.upgrade_version}}
+      - name: target-commit
+        if: ${{ inputs.upgrade_version == 'main' }}
+        run: |
+          BUILD_TAG=gaiad-linux-${{inputs.upgrade_version}}
+          TARGET_COMMIT=$(curl -s -L -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/repos/hyphacoop/cosmos-builds/releases/tags/$BUILD_TAG | jq -r '.name')
+          echo $TARGET_COMMIT
+      - name: Consumer chain binary URLs
+        run: |
+          echo "$CONSUMER_CHAIN_BINARY runs off $CONSUMER_CHAIN_BINARY_URL"
+      - name: Update PATH
+        run: |
+          echo "$HOME/go/bin" >> $GITHUB_PATH
+          echo "$HOME/.hermes" >> $GITHUB_PATH
+      - name: Bypass the grub-efi-amd64-signed package
+        run: sudo apt-mark hold grub-efi-amd64-signed
+      - name: Install required packages
+        run: |
+          sudo apt update
+          sudo apt -y install python-is-python3 python3-distutils screen curl jq wget python3-venv python3-pip
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Init Python venv
+        run: python -m venv ~/env
+      - name: Install dependencies
+        run: |
+          source ~/env/bin/activate
+          python -m pip install --upgrade pip
+          python -m pip install toml-cli
+      - name: Print RPC port
+        run: echo $VAL1_RPC_PORT
+      - name: Test software upgrade for published releases
+        if: ${{ inputs.upgrade_version != 'main' }}
+        env:
+          DOWNLOAD_URL: https://github.com/cosmos/gaia/releases/download/${{ inputs.upgrade_version }}/gaiad-${{ inputs.upgrade_version }}-linux-amd64
+        run: |
+          echo "$DOWNLOAD_URL"
+          echo "DOWNLOAD_URL=$DOWNLOAD_URL" >> $GITHUB_ENV
+      - name: Test software upgrade for main branch
+        if: ${{ inputs.upgrade_version == 'main' }}
+        env:
+          DOWNLOAD_URL: https://github.com/hyphacoop/cosmos-builds/releases/download/gaiad-linux-main/gaiad-linux
+        run: |
+          echo "$DOWNLOAD_URL"
+          echo "DOWNLOAD_URL=$DOWNLOAD_URL" >> $GITHUB_ENV
+      - name: Start chain
+        env:
+          START_VERSION: ${{inputs.gaia_version}}
+          ARCHIVE_URL: "https://files.polypore.xyz/archived-state/latest_v13.tar.gz"
+        run: tests/major_stateful_upgrade/start_chain.sh
+      - name: _Provider Block Production> gaiad v14
+        run: tests/test_block_production.sh 127.0.0.1 $VAL1_RPC_PORT 50 2100
+      - name: _Provider TX> gaiad v14
```

```suggestion
      - name: _Transactions verified> bank send, delegate, and withdraw rewards
```
Not sure if it's all three, or more than these, but we should tell people what those tx are.